### PR TITLE
Fix untar timeout issue and then some

### PR DESCRIPTION
**Pull request by jeddcuison**
Things included in this pull request

- Make read remote output a property file and set the default timeout to 180000 or 3 minutes
- Refactor code, eliminate redundant method names
- Refactor readExecRemoteOutput

### CODE REVIEW

File: jwala-common/src/main/java/com/cerner/jwala/commandprocessor/impl/jsch/JschCommandProcessorImpl.java [hunk 1]
```diff
@@ -5,19 +5,20 @@
...
+import com.cerner.jwala.common.properties.ApplicationProperties;
 import com.cerner.jwala.exception.RemoteCommandFailureException;
 
 import java.io.IOException;
 
 public class JschCommandProcessorImpl implements CommandProcessor {
 
+    private static final String JSCH_SHELL_READ_REMOTE_OUTPUT_TIMEOUT = "jsch.shell.read.remote.output.timeout";
+    private static final String JSCH_EXEC_READ_REMOTE_OUTPUT_TIMEOUT = "jsch.exec.read.remote.output.timeout";
+    private static final String DEFAULT_READ_REMOTE_OUTPUT_TIMEOUT = "180000";
```
Comment (akinny) [inline jwala-common/src/main/java/com/cerner/jwala/commandprocessor/impl/jsch/JschCommandProcessorImpl.java]: Can you add these keys to PropertyKeys, lets keep all the keys in there eventually

**Review by akinny — APPROVED** [review]

File: jwala-common/src/main/java/com/cerner/jwala/common/jsch/impl/JschServiceImpl.java [hunk 2]
```diff
@@ -50,6 +50,8 @@
     private static final int CHANNEL_EXEC_CLOSE_TIMEOUT = 300000;
     private static final String JSCH_CHANNEL_SHELL_READ_INPUT_SLEEP_DURATION = "jsch.channel.shell.read.input.sleep.duration";
```
Comment (jmahmoodcc) [inline jwala-common/src/main/java/com/cerner/jwala/common/jsch/impl/JschServiceImpl.java]: Put this one and exec read timeout constant in the PropertyKeys class

File: jwala-common/src/main/java/com/cerner/jwala/commandprocessor/impl/jsch/JschCommandProcessorImpl.java [hunk 3]
```diff
@@ -5,19 +5,20 @@
...
 import com.cerner.jwala.common.jsch.JschService;
 import com.cerner.jwala.common.jsch.RemoteCommandReturnInfo;
+import com.cerner.jwala.common.properties.ApplicationProperties;
 import com.cerner.jwala.exception.RemoteCommandFailureException;
 
 import java.io.IOException;
 
 public class JschCommandProcessorImpl implements CommandProcessor {
 
+    private static final String JSCH_SHELL_READ_REMOTE_OUTPUT_TIMEOUT = "jsch.shell.read.remote.output.timeout";
```
Comment (jmahmoodcc) [inline jwala-common/src/main/java/com/cerner/jwala/commandprocessor/impl/jsch/JschCommandProcessorImpl.java]: Same thing, let's move these to PropertyKeys

File: jwala-common/src/main/java/com/cerner/jwala/common/properties/ApplicationProperties.java [hunk 4]
```diff
@@ -99,7 +99,15 @@ public static int size() {
     }
 
     private void init() {
-        String propertiesFile = System.getProperty(PROPERTIES_ROOT_PATH) + "/" + PROPERTIES_FILE_NAME;
+
+        String propertiesRootPath = System.getProperty(PROPERTIES_ROOT_PATH);
+        if (propertiesRootPath == null) {
```
Comment (jmahmoodcc) [inline jwala-common/src/main/java/com/cerner/jwala/common/properties/ApplicationProperties.java:104]: I actually prefer forcing unit tests to declare their own Properties Root Path. I thought it was good practice not to provide default properties.

File: jwala-services/src/main/java/com/cerner/jwala/service/impl/spring/component/JschRemoteCommandExecutorServiceImpl.java [hunk 5]
```diff
@@ -15,8 +16,9 @@
 @Service
 public class JschRemoteCommandExecutorServiceImpl implements RemoteCommandExecutorService {
 
-    private static final int SHELL_REMOTE_OUTPUT_READ_WAIT_TIME = 180000;
-    private static final int EXEC_REMOTE_OUTPUT_READ_WAIT_TIME = 3000;
+    private static final String JSCH_SHELL_READ_REMOTE_OUTPUT_TIMEOUT = "jsch.shell.read.remote.output.timeout";
```
Comment (jmahmoodcc) [inline jwala-services/src/main/java/com/cerner/jwala/service/impl/spring/component/JschRemoteCommandExecutorServiceImpl.java]: Again, move to PropertyKeys

**Comment by jmahmoodcc** [comment]
Overall looks good, just some minor changes

File: jwala-common/src/main/java/com/cerner/jwala/commandprocessor/impl/jsch/JschCommandProcessorImpl.java (hunk 3 above)
Comment (akinny) [reply jwala-common/src/main/java/com/cerner/jwala/commandprocessor/impl/jsch/JschCommandProcessorImpl.java]: Don't make this change @Sharvaribarve  is making this change for the entire project

**Pull request merged**